packages/mui-lab/src/internal/pickers/hooks/usePickerState.ts

```diff
@@ -70,6 +70,8 @@ export function usePickerState<TInput, TDateValue>(
     dispatch({ type: 'reset', payload: parsedDateValue });
   }
 
+  const [initialDate, setInitialDate] = React.useState<TDateValue>(draftState.committed);
+
```

Comment:
This means `initialDate` will always be the first value of `draftState.committed`. Is this the expected behavior? (I don't know the whole implementation but creating another state kinda concern me a bit because it can create another bug where the state is not updated)

Reply:
It is set to the value of `draftState.committed` when the component first loads, and afterwards, it is newly set every time user clicks "OK". (We call setInitialDate inside acceptDate function, only if needClosePicker is true. You can check my last commit).

Reply:
I see, thanks for the explanation 👍

Reply:
Yeah, we need one more additional state :)